solana/solana-ibc/programs/solana-ibc/src/storage.rs

```diff
@@ -259,8 +259,23 @@ pub struct PrivateStorage {
     pub port_channel: map::Map<trie_ids::PortChannelPK, PortChannelStore>,
 
     pub channel_counter: u32,
+
+    pub fee_collector: Pubkey,
+
+    pub new_fee_collector_proposal: Option<Pubkey>,
+
+    pub assets: Vec<Asset>,
```

Comment:
Why not to use `Map` from denom?

Reply:
well we wont be having a lot of assets so i dont think map would do us any favor

Reply:
The `Map` is a linear map and having it will save you on having to do the lookup iteration.  It’s the same type we’re using for consensus states.

Reply:
hmm alright, i can use that